L/libnode/build_tarballs.jl

```diff
@@ -0,0 +1,72 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libnode"
+version = v"14.17.3"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://nodejs.org/dist/v14.17.3/node-v14.17.3.tar.gz", "dcbd156506ee79ee48439257626ca0a6db3d7eab8cb0208db6979125ae7d3a95"),
+]
+
+host_platform = HostPlatform()
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd node-*
+
+export CC_host=$HOSTCC
+export CXX_host=$HOSTCXX
+# Build & install libnode
+if [[ $target == $MACHTYPE ]]
+then
+    ./configure --prefix=${prefix} --shared --without-inspector --no-cross-compiling
+else
+    DEST_CPU=x86_64
+    if [[ $target == *aarch64* ]]; then DEST_CPU=arm64; fi
+    DEST_OS=linux
+    ./configure --prefix=${prefix} --shared --without-inspector --without-intl --cross-compiling --dest-cpu=$DEST_CPU --dest-os=$DEST_OS
+fi
+
+export LDFLAGS=-lrt
+export CPPFLAGS=-D__STDC_FORMAT_MACROS
+make -j`nproc`
+make install
+cp ./out/Release/node ${bindir}
+install_license ./LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# Only 64-bit Linux platforms.
+# Requires XCode and MSVC to compile on Mac OS and Windows, respectively.
```

Comment:
What XCode tools you need?  Some simple things like `xcrun` can be faked:
https://github.com/JuliaPackaging/Yggdrasil/blob/2b0395821db522fa190e749d4d7fd670a5297643/L/licensecheck/build_tarballs.jl#L17-L24

Reply:
It uses `xcrun` and `xcodebuild`, as well as something like `dtrace` which is not preinstalled on the host platform. I'm not familiar with macOS. There might be more problems to cross-compile.